dqvm/src/utilities.lisp

```diff
@@ -50,3 +50,24 @@
                 quality, consider passing a suitable seed via the command line."
                 random-seed)
     random-seed))
+
+(defvar *default-number-of-profiling-samples* 10000
+  "Default number of samples to draw during statistical profiling.")
+
+(defmacro with-profiling-maybe ((&rest package-names) &body body)
+  "Run BODY under the statistical profiler if the environment variable DQVM_PROFILE has been set to a valid profiling mode (cpu, time, or alloc). Call counts to functions in PACKAGE-NAMES are explicitly reported."
+  #+sbcl
+  (let ((profile (gensym "PROFILE-"))
+        (mode (gensym "MODE-")))
+    `(alexandria:if-let ((,profile (uiop:getenv "DQVM_PROFILE")))
+       (let ((,mode (intern (string-upcase ,profile)
+                            (find-package "KEYWORD"))))
```

Comment:
For what it's worth, there is `alexandria:make-keyword`. Not that it really buys you much :)

Reply:
I didn't know about it but I'll use it. Thank you!